src/strategy/filesystem.rs

```diff
@@ -0,0 +1,180 @@
+//! Strategy for local filesystem-based updates.
+
+use anyhow::{Context, Error, Result};
+use fn_error_context::context;
+use futures::future;
+use futures::prelude::*;
+use log::trace;
+use serde::{Deserialize, Serialize};
+use std::os::unix::fs::{MetadataExt, PermissionsExt};
+use std::pin::Pin;
+use std::time::{SystemTime, UNIX_EPOCH};
+use tokio::fs as tokio_fs;
+
+/// Struct to parse finalization marker file's JSON content into.
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct FinalizationMarker {
+    /// Unix timestamp of expiry time.
+    not_after: Option<u64>,
+}
+
+/// Strategy for immediate updates.
+#[derive(Clone, Debug, Default, Serialize)]
+pub(crate) struct StrategyFilesystem {}
+
+impl StrategyFilesystem {
+    /// Strategy label/name.
+    pub const LABEL: &'static str = "filesystem";
+    /// Local filesystem path to finalization marker file.
+    pub const FINALIZATION_MARKER_FILE_PATH: &'static str = "/var/lib/zincati/allowfinalize.json";
```

Comment:
Pending the final strategy name, I'd nest this a bit deeper so that the path hierarchy can clearly split out that this is 1) userdata input, and 2) belonging to a specific strategy.

Reply:
`/var/lib/zincati/admin/marker_file_strategy/allowfinalization.json`?

Reply:
`/var/lib/zincati/admin/strategy/<$NAME>/allowfinalization.json` I'd say.